apps/etterlatte-behandling/src/main/kotlin/behandling/domain/Behandling.kt

```diff
@@ -0,0 +1,153 @@
+            throw TilstandException.UgyldigtTilstand
+        }
+    }
+
+    open fun tilOpprettet(): Behandling {
+        throw BehandlingStoetterIkkeStatusEndringException(OPPRETTET)
+    }
+
+    open fun tilVilkaarsvurdert(utfall: VilkaarsvurderingUtfall?): Behandling {
+        throw BehandlingStoetterIkkeStatusEndringException(VILKAARSVURDERT)
+    }
+
+    open fun tilBeregnet(): Behandling {
+        throw BehandlingStoetterIkkeStatusEndringException(BEREGNET)
+    }
+
+    open fun tilFattetVedtak(): Behandling {
+        throw BehandlingStoetterIkkeStatusEndringException(FATTET_VEDTAK)
+    }
+
+    open fun tilAttestert(): Behandling {
+        throw BehandlingStoetterIkkeStatusEndringException(ATTESTERT)
+    }
+
+    open fun tilReturnert(): Behandling {
+        throw BehandlingStoetterIkkeStatusEndringException(RETURNERT)
+    }
+
+    open fun tilIverksatt(): Behandling {
+        throw BehandlingStoetterIkkeStatusEndringException(OPPRETTET)
+    }
+
+    private class BehandlingStoetterIkkeStatusEndringException constructor(
+        behandlingStatus: BehandlingStatus,
+        message: String = "Behandlingen støtter ikke statusendringen til status $behandlingStatus"
+    ) : Exception(message)
+}
+
+internal fun Behandling.toDetaljertBehandling(): DetaljertBehandling {
+    val (soeknadMottatDato, gyldighetsproeving) = when (this) {
+        is Foerstegangsbehandling -> this.soeknadMottattDato to this.gyldighetsproeving
+        // TODO øh 24.10.2022 er det riktig at søknadMottatDato er behandlingOpprettet der vi ikke har søknad?
+        else -> this.behandlingOpprettet to null
+    }
+
+    return DetaljertBehandling(
+        id = id,
+        sak = sak,
+        behandlingOpprettet = behandlingOpprettet,
+        sistEndret = sistEndret,
+        soeknadMottattDato = soeknadMottatDato,
+        innsender = persongalleri.innsender,
+        soeker = persongalleri.soeker,
+        gjenlevende = persongalleri.gjenlevende,
+        avdoed = persongalleri.avdoed,
+        soesken = persongalleri.soesken,
+        gyldighetsproeving = gyldighetsproeving,
+        status = status,
+        behandlingType = type,
+        virkningstidspunkt = this.virkningstidspunkt,
+        kommerBarnetTilgode = kommerBarnetTilgode,
+        revurderingsaarsak = when (this) {
+            is Revurdering -> revurderingsaarsak
+            else -> null
+        }
+    )
+}
+
+fun Behandling.toBehandlingSammendrag() = BehandlingSammendrag(
+    id = this.id,
+    sak = this.sak,
+    status = this.status,
+    soeknadMottattDato = if (this is Foerstegangsbehandling) {
+        this.soeknadMottattDato
+    } else {
+        this.behandlingOpprettet
+    },
+    behandlingOpprettet = this.behandlingOpprettet,
+    behandlingType = this.type,
+    aarsak = when (this) {
+        is Foerstegangsbehandling -> "SOEKNAD"
+        is Revurdering -> this.revurderingsaarsak.name
+        is ManueltOpphoer -> "MANUELT OPPHOER"
```

Comment:
Skal vi ha en enums for `aarsaker`?

Reply:
Jeg husker ikke helt hva hensikten med årsak er, så blir litt usikker på om det er verdt å gjøre noe der eller om det evt. bare skal bort.